Makefile

```diff
@@ -364,7 +347,7 @@ define generate_k8s_protobuf
 		--proto-import="$(PWD)/tools/protobuf" \
 		--packages=$(subst $(newline),$(comma),$(1)) \
 		--go-header-file "$(PWD)/hack/custom-boilerplate.go.txt" \
-		--output-base=$(2)
+		--output-dir="$(PROTOBUF_OUTPUT_DIR)"
```

Comment:
This macro now receives the `TMPDIR` as a parameter, but ignores it.

Why do we have the TMPDIR at all? I figure it's so that we can use the existing files in the tree to generate a bunch of new files in an entirely separate path, then iff all generation succeeds, we copy the full path back into the main directory. Seems a bit odd to be now generating files somewhere within the current directory instead?

Reply:
Is the base problem here that `go-to-protobuf` removed the parameters we're relying on and refuses to recreate the directory structure in the output dir based on the input directory structure, thereby forcing us to directly inject the files back into the same tree?

Reply:
Ugh yeah looks like upstream broke the ability to output somewhere outside `$GOPATH`. I see other commands adding a `--output-package` to provide the equivalent GOPATH path presumably to infer something about the output, but they didn't bother for this command.

The best I'm coming up with so far as an alternative here is accept dirtying the tree and use this instead:
```
    --output-dir=$$GOPATH/src
```

I'll carry this change in https://github.com/cilium/cilium/pull/32063 since it'll be a dependency there if you don't integrate it in before merge.